crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1748,8 +1748,37 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.infer_arguments(arguments);
         self.infer_expression(func);
 
-        // TODO resolve to return type of `func`, if its a callable type
-        Type::Unknown
+        match func.as_ref() {
+            name_expr @ ast::Expr::Name(..) => {
+                let name_expr_id = name_expr.scoped_ast_id(self.db, self.scope);
+                let Some(named_type) = self.types.expressions.get(&name_expr_id) else {
+                    return Type::Unbound;
+                };
+
+                match named_type {
+                    Type::Function(function_type) => {
+                        function_type.returns(self.db).unwrap_or(Type::Unknown)
+                    }
+
+                    // TODO: handle class constructors
+                    // TODO: handle classes which implement the Callable protocol
+                    Type::Class(_class_type) => Type::Unknown,
+
+                    // `Any` is callable, and its return type is also `Any`.
+                    Type::Any => Type::Any,
+
+                    // TODO: union and intersection types, if they reduce to `Callable`?
+                    Type::Union(_) => todo!(),
+                    Type::Intersection(_) => todo!(),
```

Comment:
We shouldn't use `todo!()` in cases we aren't handling yet, because this will result in panics analyzing some code, which we want to avoid. The pattern to use for things we don't handle yet is `Type::Unknown` plus a `TODO` comment.

Reply:
Doh! I thought I’d updated all of those. Thank you. 👍🏼